File: LeetCode/[E] - 1431. Kids With the Greatest Number of Candies/solution.go
package main

func kidsWithCandies(candies []int, extraCandies int) []bool {
	var maxCand int
	for _, el := range candies {
		if el > maxCand {
			maxCand = el
		}
	}
	res := make([]bool, 0, len(candies))
	for _, el := range candies {
		if el+extraCandies >= maxCand {
			res = append(res, true)
		} else {
			res = append(res, false)
		}
	}
	return res
}

File: LeetCode/[E] - 1431. Kids With the Greatest Number of Candies/solution.py
class Solution:
    def kidsWithCandies(
        self, candies: list[int], extraCandies: int
    ) -> list[bool]:
        max_cand = max(candies)
        res = []
        for cand in candies:
            res.append([False, True][cand + extraCandies >= max_cand])
        return res
